cmd/serve_test.go

```diff
@@ -21,13 +20,18 @@ var _ = Describe("Serve command", func() {
 
 			isConfigMandatory = false
 
+			grClosure := make(chan interface{})
```

Comment:
To be honest, I don't understand my own test. I think the idea was to start the server (without errors) and to shut them down (by sending the signal via "done" channel). But the server starts with error because default upstream is not defined. :confused: 

I think your change is OK here, because the server will not block (since it starts with error).

Reply:
yea seams like it but caused flaky behavior in the following tests as the go routine might still be running depending on the performance & context switches of the testing machine.
I tried to implement th go routine handling acording to [gomega](https://onsi.github.io/gomega/#making-asynchronous-assertions).